### PR TITLE
linux config: enable Landlock LSM

### PR DESCRIPTION
**Pull request by Lunarequest**
###### Motivation for this change
Since kernel 5.13 landlock lsm has been merged into mainline. This allows unprivileged restriction of ambient rights (e.g. global filesystem access) for a set of processes. An example of this can be seen in the gnu tar [patch set](https://github.com/landlock-lsm/tar/tree/landlock-v1) for landlock lsm support. This will be a
###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [x] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [21.11 Release Notes (or backporting 21.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2111-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by 06kellyjac** [comment]
Can you squash this into one commit and correct your commit and PR title to follow [contributing](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md) please

**Comment by 06kellyjac** [comment]
Admittedly the commits to this file are a bit hit and miss but `linux: something` seems most common
https://github.com/NixOS/nixpkgs/commits/master/pkgs/os-specific/linux/kernel/common-config.nix

**Comment by advaithm** [comment]
I've based the commit message and pr name on the pr that enabled the Yama lsm, is it fine @06kellyjac

File: pkgs/os-specific/linux/kernel/common-config.nix [hunk 1]
```diff
@@ -443,6 +443,9 @@ let
       SECURITY_SELINUX_BOOTPARAM_VALUE = whenOlder "5.1" (freeform "0"); # Disable SELinux by default
       # Prevent processes from ptracing non-children processes
       SECURITY_YAMA                    = option yes;
+      # The goal of Landlock is to enable to restrict ambient rights (e.g. global filesystem access) for a set of processes. this does not have any effect if a
+      # program does not support it
```
Comment (SuperSandro2000) [inline pkgs/os-specific/linux/kernel/common-config.nix:447]:
```suggestion
      # The goal of Landlock is to enable to restrict ambient rights (e.g. global filesystem access) for a set of processes.
      # This does not have any effect if a program does not support it
```

**Comment by Lunarequest** [comment]
Is there any update on this?

**Comment by SuperSandro2000** [comment]
The epson driver commit you added by accident belongs into another PR.

**Comment by Lunarequest** [comment]
> The epson driver commit you added by accident belongs into another PR.

sorry will remove

**Comment by SuperSandro2000** [comment]
Can you squash the three commits or someone squash merge the commit?

**Comment by Lunarequest** [comment]
> Can you squash the three commits or someone squash merge the commit?

Could someone do it, i'm not to sure i can pull it off

**Comment by SuperSandro2000** [comment]
You didn't have maintainer push activated so I pushed to a new branch. #155606

**Comment by Lunarequest** [comment]
> You didn't have maintainer push activated so I pushed to a new branch. #155606

Oh thank you, i'll close this in favor of your pr

**Pull request closed without merging**